### ISSUE TITLE
[BUG] Custom ESLINT config is checking files which are not relevant

### ISSUE THREAD

**Issue by DougMidgley** [forcedotcom/sfdx-scanner#1146]
**Describe the bug**
Running SFDX Scanner ESLINT just custom config is throwing parsing errors on files such as .component or .json unless explicitly added to ESLINT ignore. This does not happen when running ESLINT directly

**To Reproduce**
run the following on a folder which contains .component or .json files. where you have a custom eslint config.

` sfdx scanner:run -t force-app --eslintconfig ".eslintrc.json" -o codescan/code-scan.csv -f csv -s 3`

outcome is for example
```

{
    "engine": "eslint-custom",
    "fileName": "******\\force-app\\b2b\\lwc\\b2bCheckPasswordResetEmail\\__tests__\\data\\CurrentPageReference.json",
    "violations": [
      {
        "line": 2,
        "column": 11,
        "severity": 2,
        "message": "Parsing error: Unexpected token :",
        "ruleName": "files-must-compile",
        "category": "Scanner Internal",
        "url": "",
        "normalizedSeverity": 1
      }
    ]
  }
```

**Expected behavior**

I would expect parity between eslint and sfdx scanner eslint to avoid unxpected issues and improve developer experience. IF eslint is parsing JSON, this should be ok, but if not it should also not in SFDX scanner eslint.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Scanner Version 3.15

**Additional context**

likely linked to https://github.com/forcedotcom/sfdx-scanner/blob/4b5405363e28193493afcf86dffe04fcff01860e/src/lib/eslint/CustomEslintEngine.ts#L60

"Workaround": 
- using Target flag with only .js in path - means cannot run both PMD and JS easily
- adding all irrelevant types to eslint ignore key - massive maintenance effort

"Urgency": "Not business-stopping".

**Comment by jfeingold35** [comment]
@DougMidgley , thanks for bringing this to our attention. I know that there's more context for this on a previous issue, but in order to simplify our bookkeeping, could you please add steps to reproduce, as well as the other information requested in the Issue template?
Thanks!

**Comment by DougMidgley** [comment]
sure thing :) created from the code line mentioned so seems it doesnt set template :)

**Comment by stephen-carter-at-sf** [comment]
I think what we can do here is make the eslint-custom engine have a config inside of ~/.sfdx-scanner/Config.json with its default target files which I think should be equal to the union of what the other eslint-* engines do which would be:
			targetPatterns: [
				"**/*.js",
				"**/*.ts",
				"!**/node_modules/**",
				"!**/bower_components/**"
			]
And then if user wants to include/exclude other files - they can update their Config file.

**Comment by stephen-carter-at-sf** [comment]
Will be fixed in 4.1.0 which should release and the end of this month.
